Replace current Base64

## Description

Review the usage of Base64 because it throws a warning in the Pipeline:

``` 
/home/runner/work/amadeus-java/amadeus-java/src/main/java/com/amadeus/travel/TripParser.java:9: warning: Base64 is internal proprietary API and may be removed in a future release
import com.sun.org.apache.xerces.internal.impl.dv.util.Base64;
```

Analyse to replace with Base64:

https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html

Juan Antonio

--- a/src/main/java/com/amadeus/travel/TripParser.java
+++ b/src/main/java/com/amadeus/travel/TripParser.java
@@ -88,15 +88,16 @@ public class TripParser {
    */
   public TripDetail post(File file) throws ResponseException, IOException {
     // Base64 encode file and create request body
-    FileInputStream fileInputStreamReader = new FileInputStream(file);
-    byte[] bytes = new byte[(int)file.length()];
-    fileInputStreamReader.read(bytes);
-    String encodedFile = Base64.getEncoder().encodeToString(bytes);
-    JsonObject body = new JsonObject();
-    body.addProperty("payload", encodedFile);
+    try (FileInputStream fileInputStreamReader = new FileInputStream(file)) {
+      byte[] bytes = new byte[(int)file.length()];
+      fileInputStreamReader.read(bytes);
+      String encodedFile = Base64.getEncoder().encodeToString(bytes);
+      JsonObject body = new JsonObject();
+      body.addProperty("payload", encodedFile);
 
-    Response response = client.post("/v3/travel/trip-parser", body);
-    return (TripDetail) Resource.fromObject(response, TripDetail.class);
+      Response response = client.post("/v3/travel/trip-parser", body);
+      return (TripDetail) Resource.fromObject(response, TripDetail.class);
+    }
   }
 
   /**

--- a/src/main/java/com/amadeus/travel/TripParser.java
+++ b/src/main/java/com/amadeus/travel/TripParser.java
@@ -6,11 +6,10 @@ import com.amadeus.exceptions.ResponseException;
 import com.amadeus.resources.Resource;
 import com.amadeus.resources.TripDetail;
 import com.google.gson.JsonObject;
-import com.sun.org.apache.xerces.internal.impl.dv.util.Base64;
-
 import java.io.File;
+import java.io.FileInputStream;
 import java.io.IOException;
-import java.nio.file.Files;
+import java.util.Base64;
 
 /**
  * <p>
@@ -89,9 +88,12 @@ public class TripParser {
    */
   public TripDetail post(File file) throws ResponseException, IOException {
     // Base64 encode file and create request body
-    String b64Encoded = Base64.encode(Files.readAllBytes(file.toPath()));
+    FileInputStream fileInputStreamReader = new FileInputStream(file);
+    byte[] bytes = new byte[(int)file.length()];
+    fileInputStreamReader.read(bytes);
+    String encodedFile = Base64.getEncoder().encodeToString(bytes);
     JsonObject body = new JsonObject();
-    body.addProperty("payload", b64Encoded);
+    body.addProperty("payload", encodedFile);
 
     Response response = client.post("/v3/travel/trip-parser", body);
     return (TripDetail) Resource.fromObject(response, TripDetail.class);

--- a/src/main/java/com/amadeus/travel/TripParser.java
+++ b/src/main/java/com/amadeus/travel/TripParser.java
@@ -90,14 +90,19 @@ public class TripParser {
     // Base64 encode file and create request body
     try (FileInputStream fileInputStreamReader = new FileInputStream(file)) {
       byte[] bytes = new byte[(int)file.length()];
-      fileInputStreamReader.read(bytes);
-      String encodedFile = Base64.getEncoder().encodeToString(bytes);
-      JsonObject body = new JsonObject();
-      body.addProperty("payload", encodedFile);
+      int count = 0;
+      if (fileInputStreamReader.read(bytes) > 0) {
+        String encodedFile;
+        encodedFile = Base64.getEncoder().encodeToString(bytes);
+        JsonObject body = new JsonObject();
+        body.addProperty("payload", encodedFile);
+        count = count + fileInputStreamReader.read(bytes);
 
-      Response response = client.post("/v3/travel/trip-parser", body);
-      return (TripDetail) Resource.fromObject(response, TripDetail.class);
+        Response response = client.post("/v3/travel/trip-parser", body);
+        return (TripDetail) Resource.fromObject(response, TripDetail.class);
+      }
     }
+    return null;
   }
 
   /**

